shared/src/main/scala/mlscript/NewParser.scala

```diff
@@ -1173,6 +1173,10 @@ abstract class NewParser(origin: Origin, tokens: Ls[Stroken -> Loc], newDefs: Bo
         consume
         consume
         S(Var(idStr).withLoc(S(l0)))
+      case (LITVAL(IntLit(i)), l0) :: (KEYWORD(":"), _) :: _ if inRecord => // TODO: | ...
```

Comment:
What's the rationale for only allowing the parser to parse this in records? It complicates the parser for no apparent reason. Just parse things uniformly and let the semantic analysis deal with it later, which will also produce better errors.

Reply:
Alright. Actually, I also feel that adding an `inRecord` parameter everywhere just for parsing this specific syntax is quite clumsy. Let me think about how to do this…

Reply:
I recall that the reason I did this is it just mixes up ascription in argument lists with named arguments. 

```
foo(123 : Int) : Int
//│ Int
//│ res
//│     = 124

foo(123: Int): Int
//│ ╔══[ERROR] Cannot use named arguments as the function type has untyped arguments
//│ ║  l.34: 	foo(123: Int): Int
//│ ╙──      	   ^^^^^^^^^^
//│ Int
//│ Code generation encountered an error:
//│   unresolved symbol Int

foo(123:Int):Int
//│ ╔══[ERROR] Cannot use named arguments as the function type has untyped arguments
//│ ║  l.36: 	foo(123:Int):Int
//│ ╙──      	   ^^^^^^^^^
//│ Int
//│ Code generation encountered an error:
//│   unresolved symbol Int
```

I see. I know how to handle this with semantic analysis.

Reply:
You're just hitting a current stupidly confusing "feature" of the parser that `a: b` is treated as a name-value association while `a : b` is treated as a type ascription. This has nothing to do with integer-named fields AFAIK.

Reply:
Thank you for your clarification. I believe I also need to add `:e` to test cases using terms like `f(42: Int)` to track this change.

Reply:
I have updated the new parser in the latest commit.